grakn-engine/src/main/java/ai/grakn/engine/task/postprocessing/RedisIndexStorage.java

```diff
@@ -0,0 +1,104 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.task.postprocessing;
+
+import ai.grakn.Keyspace;
+import ai.grakn.concept.ConceptId;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.annotations.VisibleForTesting;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.Response;
+import redis.clients.jedis.Transaction;
+import redis.clients.util.Pool;
+
+import javax.annotation.Nullable;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ *
+ * <p>
+ *     Stores a list of indices and vertex ids representing those indices which need to be post processed
+ * </p>
+ *
+ * @author fppt
+ */
+public class RedisIndexStorage extends RedisStorage {
+    private RedisIndexStorage(Pool<Jedis> jedisPool, MetricRegistry metricRegistry) {
+        super(jedisPool, metricRegistry);
+    }
+
+    public static RedisIndexStorage create(Pool<Jedis> jedisPool, MetricRegistry metricRegistry) {
+        return new RedisIndexStorage(jedisPool, metricRegistry);
+    }
+
+    /**
+     * Add an index to the list of indices which needs to be post processed
+     */
+    public void addIndex(Keyspace keyspace, String index, Set<ConceptId> conceptIds){
+        String listOfIndicesKey = getIndicesKey(keyspace);
+        String listOfIdsKey = getConceptIdsKey(keyspace, index);
+
+        contactRedis(jedis -> {
+            //Track all the indices which need to be post proceed
+            jedis.sadd(listOfIndicesKey, index);
+            conceptIds.forEach(id -> jedis.sadd(listOfIdsKey, id.getValue()));
+            return null;
+        });
+    }
+
+    /**
+     * Gets and removes the next index to post process
+     */
+    @Nullable
+    public String popIndex(Keyspace keyspace){
+        String indexKey = getIndicesKey(keyspace);
+        return contactRedis(jedis -> jedis.spop(indexKey));
+    }
+
+    /**
+     * Gets and removes all the ids which we need to post process
+     */
+    public Set<ConceptId> popIds(Keyspace keyspace, String index){
+        String idKey = getConceptIdsKey(keyspace, index);
+        return contactRedis(jedis -> {
+            Transaction tx = jedis.multi();
+            Response<Set<String>> responseIds = tx.smembers(idKey);
+            tx.del(idKey);
+            tx.exec();
+            return  responseIds.get().stream().map(ConceptId::of).collect(Collectors.toSet());
+        });
+    }
+
+    /**
+     * The key which refers to  a list of all the indices in a certain {@link Keyspace} which need to be post processed
+     */
+    @VisibleForTesting
+    static String getIndicesKey(Keyspace keyspace){
+        return "IndicesToProcess_" + keyspace.getValue();
+    }
```

Comment:
I would be tempted to just hard-code or duplicate this logic in the tests instead of exposing it. After all, we want to check the keys look like how we expect (e.g. that they contain the keyspace)